<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="R interface to search orthologs between species, using the [DIOPT](https://www.flyrnai.org/diopt) tool.">
<title>Query Ortholog using DIOPT service • RDIOPT</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Query Ortholog using DIOPT service">
<meta property="og:description" content="R interface to search orthologs between species, using the [DIOPT](https://www.flyrnai.org/diopt) tool.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">RDIOPT</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="rdiopt">RDIOPT<a class="anchor" aria-label="anchor" href="#rdiopt"></a>
</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>RDIOPT is a simple wrapper around the <a href="https://www.flyrnai.org/diopt" class="external-link">DIOPT</a> Ortholog Finder. The intention for this package is to allow querying orthologs from R directly, without copy and paste.</p>
<p>RDIOPT relies on the <a href="https://www.flyrnai.org/tools/diopt/web/api" class="external-link">DIOPT API</a> to query and search results.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of RDIOPT from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"myuhao/RDIOPT"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>Call <code><a href="reference/find_orthologs.html">find_orthologs()</a></code> with some of your gene of interests. The function should give you a nice <code>tibble</code> with all the relevant information.</p>
<div>
<blockquote>
<p><strong>Note</strong></p>
<p>Note that currently DIOPT API only supports <strong>Entrez ID</strong>. You will need to convert your genes accordingly.</p>
</blockquote>
</div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RDIOPT</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gene_of_interests</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1232</span>, <span class="fl">7316</span><span class="op">)</span></span>
<span><span class="va">orthologs</span> <span class="op">=</span> <span class="fu"><a href="reference/find_orthologs.html">find_orthologs</a></span><span class="op">(</span><span class="va">gene_of_interests</span>, from <span class="op">=</span> <span class="fl">9606</span>, to <span class="op">=</span> <span class="fl">7227</span>, filter <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">orthologs</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "date"                         "search_gene_entrez"          </span></span>
<span><span class="co">#&gt;  [3] "filter"                       "from_entrez_geneid"          </span></span>
<span><span class="co">#&gt;  [5] "symbol"                       "description"                 </span></span>
<span><span class="co">#&gt;  [7] "chromosome"                   "gene_type"                   </span></span>
<span><span class="co">#&gt;  [9] "ensembl_geneid"               "diopt_version"               </span></span>
<span><span class="co">#&gt; [11] "to_id"                        "to_symbol"                   </span></span>
<span><span class="co">#&gt; [13] "from_id"                      "confidence"                  </span></span>
<span><span class="co">#&gt; [15] "score"                        "best_score"                  </span></span>
<span><span class="co">#&gt; [17] "max_score"                    "best_score_rev"              </span></span>
<span><span class="co">#&gt; [19] "best_score_count"             "mist_ppi"                    </span></span>
<span><span class="co">#&gt; [21] "mist_genetic"                 "to_entrez_geneid"            </span></span>
<span><span class="co">#&gt; [23] "species_id"                   "species_specific_geneid"     </span></span>
<span><span class="co">#&gt; [25] "species_specific_geneid_type" "count"                       </span></span>
<span><span class="co">#&gt; [27] "methods"</span></span></code></pre></div>
<p>The result table has many relevant information, but it can be summarized in to three categories:</p>
<ol style="list-style-type: decimal">
<li>Function parameters.</li>
<li>Source gene information.</li>
<li>Target gene information.</li>
</ol>
<div class="section level4">
<h4 id="function-parameters">Function parameters<a class="anchor" aria-label="anchor" href="#function-parameters"></a>
</h4>
<p>DIOPT recorded the following parameter used for the query:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 7 × 4</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   date              search_gene_entrez filter diopt_version</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                          &lt;int&gt; &lt;chr&gt;          &lt;int&gt;</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 22-10-20 22:17:56               1232 none               8</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 22-10-20 22:17:56               7316 none               8</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 22-10-20 22:17:56               7316 none               8</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 22-10-20 22:17:56               7316 none               8</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 22-10-20 22:17:56               7316 none               8</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 22-10-20 22:17:56               7316 none               8</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7 22-10-20 22:17:56               7316 none               8</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="source-gene-information">Source gene information<a class="anchor" aria-label="anchor" href="#source-gene-information"></a>
</h4>
<p>DIOPT will try to find information of the gene you submitted.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 7 × 6</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   from_entrez_geneid symbol description                  chrom…¹ gene_…² ensem…³</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                &lt;int&gt; &lt;chr&gt;  &lt;chr&gt;                          &lt;int&gt; &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1               1232 CCR3   C-C motif chemokine recepto…       3 protei… ENSG00…</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2               7316 UBC    ubiquitin C                       12 protei… ENSG00…</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3               7316 UBC    ubiquitin C                       12 protei… ENSG00…</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4               7316 UBC    ubiquitin C                       12 protei… ENSG00…</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5               7316 UBC    ubiquitin C                       12 protei… ENSG00…</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6               7316 UBC    ubiquitin C                       12 protei… ENSG00…</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7               7316 UBC    ubiquitin C                       12 protei… ENSG00…</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with abbreviated variable names ¹​chromosome, ²​gene_type, ³​ensembl_geneid</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="source-gene-information-1">Source gene information<a class="anchor" aria-label="anchor" href="#source-gene-information-1"></a>
</h4>
<p>Here is the actual ortholog information. Fileds are left <code>NA</code> in case no orthologs can be found.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 7 × 17</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   to_id  to_symbol from_id confi…¹ score best_…² max_s…³ best_…⁴ best_…⁵ mist_…⁶</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;   &lt;int&gt; &lt;chr&gt;     &lt;int&gt; &lt;chr&gt;     &lt;int&gt; &lt;chr&gt;  </span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 &lt;NA&gt;   No Ortho… &lt;NA&gt;    &lt;NA&gt;       NA &lt;NA&gt;         NA &lt;NA&gt;         NA &lt;NA&gt;   </span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 38456  Ubi-p63E  7316    high       11 Yes          15 Yes           2 -      </span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 326237 Ubi-p5E   7316    modera…     5 No           15 Yes           1 -      </span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 31564  CG11700   7316    low         2 No           15 No            0 -      </span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 33629  RpL40     7316    low         1 No           15 No            0 -      </span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 34420  RpS27A    7316    low         1 No           15 No            0 -      </span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7 35151  Nedd8     7316    low         1 No           15 No            0 -      </span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 7 more variables: mist_genetic &lt;chr&gt;, to_entrez_geneid &lt;int&gt;,</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   species_id &lt;int&gt;, species_specific_geneid &lt;chr&gt;,</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   species_specific_geneid_type &lt;chr&gt;, count &lt;int&gt;, methods &lt;chr&gt;, and</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   abbreviated variable names ¹​confidence, ²​best_score, ³​max_score,</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ⁴​best_score_rev, ⁵​best_score_count, ⁶​mist_ppi</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing RDIOPT</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Yuhao Min <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Yuhao Min.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
